app/models/chef.rb

```diff
@@ -1,4 +1,10 @@
 class Chef < ApplicationRecord
    validates_presence_of :name
-   has_many :dishes
+   
+   has_many :dishes, dependent: :destroy
+   has_many :ingredients, through: :dishes
+
+   def list_ingredients
+      ingredients.pluck(:name)
```

Comment:
Nice work! Adding the additional association makes this logic much easier. But, make sure you add `.distinct` to make it unique